### PR TITLE
change: single_include: WinCE compatibility.

### PR DESCRIPTION
**Pull request by blackhearted**
Made single_include/catch.hpp buildable under WinCE.
Changes:
- Coloring is not supported
- ::OutputDebugStringA -> ::OutputDebugString
- std::time() -> own time() implementation

### CODE REVIEW

File: single_include/catch.hpp [hunk 1]
```diff
@@ -6817,7 +6851,8 @@ namespace {
         static Win32ColourImpl s_instance;
         return &s_instance;
     }
-
+#else
```
Comment (nabijaczleweli) [inline single_include/catch.hpp]: Empty `#else` clause any reason it's there?

File: single_include/catch.hpp [hunk 2]
```diff
@@ -7699,10 +7742,18 @@ namespace Catch {
 #endif // Platform
 
 #ifdef CATCH_PLATFORM_WINDOWS
-    extern "C" __declspec(dllimport) void __stdcall OutputDebugStringA( const char* );
+#ifndef _WIN32_WCE//WinCE compatibility
+    extern "C" __declspec(dlliport) void __stdcall OutputDebugStringA( const char* );
```
Comment (janisozaur) [inline single_include/catch.hpp]: typo in `dlliport`, missing `m`

File: single_include/catch.hpp (hunk 1 above)
Comment (blackhearted) [inline single_include/catch.hpp]: No, just weird part of coding convention I do not always follow.

On 8 January 2016 at 13:51, Jędrzej <notifications@github.com> wrote:

> In single_include/catch.hpp
> <https://github.com/philsquared/Catch/pull/570#discussion_r49185040>:
>
> > @@ -6817,7 +6851,8 @@ namespace {
> >          static Win32ColourImpl s_instance;
> >          return &s_instance;
> >      }
> > -
> > +#else
>
> Empty #else clause any reason it's there?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/philsquared/Catch/pull/570/files#r49185040>.
>



-- 
Best Regards,
Vladimir Stepanov

**Comment by horenmar** [comment]
Changes should go into the right header inside `include/internal` folder, not into the single-include header (single-include header is updated on releases).

Since there is no way for me to test this, I won't be splitting it out, but if you are still around and interested, I think it can get into master, as a provisional support for WinCE.

**Comment by horenmar** [comment]
There has been no followup, so I am going to close this.

**Pull request closed without merging**